istioctl/cmd/istioctl/istioctl_test.go

```diff
@@ -568,3 +576,129 @@ func TestVersion(t *testing.T) {
+			wantException:  true, // "istioctl get invalid" should fail
+		},
+		{ // case 4 routes invalid
+			configs:        []model.Config{},
+			args:           strings.Split("proxy-config routes invalid", " "),
+			expectedRegexp: regexp.MustCompile("^Error: unable to retrieve Pod: pods \"invalid\" not found.*"),
+			wantException:  true, // "istioctl get invalid" should fail
+		},
+		{ // case 5 bootstrap invalid
+			configs:        []model.Config{},
+			args:           strings.Split("proxy-config bootstrap invalid", " "),
+			expectedRegexp: regexp.MustCompile("^Error: unable to retrieve Pod: pods \"invalid\" not found.*"),
+			wantException:  true, // "istioctl get invalid" should fail
+		},
+		{ // case 6 clusters valid
+			configs:          []model.Config{},
+			execClientConfig: cannedConfig,
+			args:             strings.Split("proxy-config clusters details-v1-5b7f94f9bc-wp5tb", " "),
+			expectedOutput: `SERVICE FQDN                                    PORT      SUBSET     DIRECTION     TYPE
+istio-policy.istio-system.svc.cluster.local     15004     -          outbound      EDS
+xds-grpc                                        -         -          -             STRICT_DNS
+`,
+		},
+		{ // case 7 listeners valid
+			configs:          []model.Config{},
+			execClientConfig: cannedConfig,
+			args:             strings.Split("proxy-config listeners details-v1-5b7f94f9bc-wp5tb", " "),
+			expectedOutput: `ADDRESS            PORT     TYPE
+172.21.134.116     443      TCP
+0.0.0.0            8080     HTTP
+`,
+		},
+		{ // case 8 routes valid
+			configs:          []model.Config{},
+			execClientConfig: cannedConfig,
+			args:             strings.Split("proxy-config routes details-v1-5b7f94f9bc-wp5tb", " "),
+			expectedOutput: `NOTE: This output only contains routes loaded via RDS.
+NAME                                                    VIRTUAL HOSTS
+15004                                                   2
+inbound|9080||productpage.default.svc.cluster.local     1
+`,
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case %d %s", i, strings.Join(c.args, " ")), func(t *testing.T) {
+			verifyOutput(t, c)
+		})
+	}
+}
+
+func TestProxyStatus(t *testing.T) {
+	cases := []testCase{
+		{ // case 0
+			configs:        []model.Config{},
+			args:           strings.Split("proxy-status", " "),
+			expectedRegexp: regexp.MustCompile("^PROXY     CDS     LDS     EDS     RDS     PILOT"),
+		},
+		{ // case 1 short name "ps"
+			configs:        []model.Config{},
+			args:           strings.Split("ps", " "),
+			expectedRegexp: regexp.MustCompile("^PROXY     CDS     LDS     EDS     RDS     PILOT"),
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case %d %s", i, strings.Join(c.args, " ")), func(t *testing.T) {
+			verifyOutput(t, c)
+		})
+	}
+}
+
+func mockClientExecFactoryGenerator(testResults map[string][]byte) func(kubeconfig, configContext string) (kubernetes.ExecClient, error) {
```

Comment:
This feels like overkill for just creating a `MockExecClient` struct. This is why I'm not a huge fan of factories in Go.

Not a blocking comment though.